frontends/p4/simplifySwitch.cpp

```diff
@@ -41,9 +41,15 @@ bool DoSimplifySwitch::matches(const IR::Expression* left, const IR::Expression*
 const IR::Node* DoSimplifySwitch::postorder(IR::SwitchStatement* stat) {
     if (!typeMap->isCompileTimeConstant(stat->expression))
         return stat;
+    bool foundMatch = false;
     for (auto ss : stat->cases) {
-        if (matches(ss->label, stat->expression))
+        if (matches(ss->label, stat->expression)){
```

Comment:
so it looks like this was a bug.
Would be great if there was a test case for this bug.

Reply:
missing a space here